nengo/utils/progress.py

```diff
@@ -390,42 +609,47 @@ def update(self, progress):
 
 
 class ProgressTracker(object):
-    """Tracks the progress of some process with a progress bar.
-
-    Parameters
-    ----------
-    max_steps : int
-        Maximum number of steps of the process.
-    progress_bar : :class:`ProgressBar` or :class:`ProgressUpdater`
-        The progress bar to display the progress.
-    """
-    def __init__(self, max_steps, progress_bar, task):
-        self.progress = Progress(max_steps)
-        self.progress_bar = wrap_with_progressupdater(
-            task=task, progress_bar=progress_bar)
+    def __init__(self, progress_bar, total_progress, update_interval=0.1):
+        self.progress_bar = to_progressbar(progress_bar)
+        self.total_progress = total_progress
+        self.update_interval = update_interval
+        self.update_thread = threading.Thread(target=self.update_loop)
+        self.update_thread.daemon = True
+        self._closing = False
+        self.sub_progress = None
+
+    def subtask(self, max_steps=None, ongoing_name='', finished_name=None):
```

Comment:
Found this function name a little bit confusing. It's not just accessing the currently running subtask, it's creating a new subtask to track the progress of. Maybe rename it `next_subtask` or `inc_subtask` or `new_subtask` or `get_subtask`? I know naming things is hard, but I thought it might be worth mentioning.